pkg/controller/external_tfpluginfw.go

```diff
@@ -254,33 +254,34 @@ func (n *terraformPluginFrameworkExternalClient) getDiffPlan(ctx context.Context
 	}
 	planResponse, err := n.server.PlanResourceChange(ctx, prcReq)
 	if err != nil {
-		return &tfprotov5.DynamicValue{}, false, errors.Wrap(err, "cannot plan change")
+		return nil, false, errors.Wrap(err, "cannot plan change")
 	}
 	if fatalDiags := getFatalDiagnostics(planResponse.Diagnostics); fatalDiags != nil {
-		return &tfprotov5.DynamicValue{}, false, errors.Wrap(fatalDiags, "plan resource change request failed")
-	}
-
-	if len(planResponse.RequiresReplace) > 0 {
-		var sb strings.Builder
-		sb.WriteString("diff contains fields that require resource replacement: ")
-		for _, attrPath := range planResponse.RequiresReplace {
-			sb.WriteString(attrPath.String())
-			sb.WriteString(", ")
-		}
-		return nil, false, errors.New(sb.String())
+		return nil, false, errors.Wrap(fatalDiags, "plan resource change request failed")
 	}
 
 	plannedStateValue, err := planResponse.PlannedState.Unmarshal(n.resourceValueTerraformType)
 	if err != nil {
 		return nil, false, errors.Wrap(err, "cannot unmarshal planned state")
 	}
 
-	diffso, err := plannedStateValue.Diff(tfStateValue)
+	rawDiff, err := plannedStateValue.Diff(tfStateValue)
 	if err != nil {
 		return nil, false, errors.Wrap(err, "cannot compare prior state and plan")
 	}
 
-	return planResponse.PlannedState, len(diffso) > 0, nil
+	// filter diffs that has unknown plan value which corresponds to
+	// computed values. These cause unnecessary diff detection when only computed
+	// attribute diffs exist in the raw diff and no actual diff exists in the
+	// parametrizable attributes
+	filteredDiff := make([]tftypes.ValueDiff, 0)
+	for _, diff := range rawDiff {
+		if diff.Value1.IsKnown() {
```

Comment:
For understanding better, as I understand, there are two `*Value` in `ValueDiff`: `Value1` and `Value2`. We use `Value1` here. Do we have an assumption here? Is it possible that we are missing something by not looking at Value2 at this point?

Reply:
hi, we are comparing planned state with current state . Therefore in the resulting diff `Value1` corresponds to plan value and `Value2` is current state value (since invoked as `plannedStateValue.Diff(tfStateValue)` , plan is at LHS, current state at RHS).

Current state values are always known by nature, since all is read from cloud API. Plan can contain unknown values, since there can be computed values assigned by the cloud API, which can only be known after applying them. 

For such attributes we are not interested in their diffs when they are the only ones, i.e the calculated diff only consists of computed attributes (observable fields) with unknown values and no parametrizable field has changed.